scripts/architecture_ownership_check.py

```
@@ -16,23 +16,64 @@
+
+
+def _check_readme(errors: list[str]) -> None:
+    if not README.is_file():
+        errors.append("missing README.md")
+        return
+    text = README.read_text(encoding="utf-8")
+    for marker in README_REQUIRED_MARKERS:
+        if marker not in text:
+            errors.append(f"README.md missing required marker: {marker!r}")
+    # Forbidden production-UI claims
+    if "Vite :5173" in text or "Vite/Caddy SPA" in text and "not a Vite" not in text:
```

Ruff (0.15.21)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)